### ISSUE TITLE
List of pandas dataframes in variable explorer slows down Spyder

### ISSUE THREAD

**Issue by a-seyler** [spyder-ide/spyder#2560]
Hello,

I'm loading a list of pandas dataframes into the workspace. When the variable explorer is open and contains this list as a variable, spyder gets really slow (It then takes several seconds to switch from editor to the ipython console by clicking into it for example). When I switch to the file explorer or the object inspector and the variable explorer is not visible, everything works fine and as fast as usual. When I switch back to the variable explorer, Spyder reacts very slowly again. When I change the name of this list into uppercase letters and activate the option "Exclude capitalized references" so that this variable is not visible in the variable explorer, Spyder works fine. I see the same behaviour  when a dict of dataframes is in the workspace. This problem arised when I updated my anaconda distribution ("conda update conda") a few days ago and was not the case with some older version I used before. I'm now using Spyder version 2.3.5.2, pandas 0.16.2 and numpy 1.9.2 on Windows Server 2008R2 (x64). 

Here is a small code example:

```python
import pandas as pd
import numpy as np
df = pd.DataFrame(np.random.randn(100, 10))
list_of_dfs = [df] * 100
```

When there are only a few dataframes in the list (e.g. 10), this is barely noticeable. But by increasing the number the slowdown gets worse. However, clicking onto the list in the variable explorer to view it's content is surprisingly fast.

**Comment by ccordoba12** [comment]
Thanks a lot for reporting, I'll see what's happening for our next release

**Comment by ryscet** [comment]
I have the exact same problem. I store a lot of pandas dataframes (2D gaze signal slices) in a list. When this list is exposed in variable explorer switching between it and the console takes significant amount of time (couple of seconds on Mac os, 24 GB ram, 3.46 Ghz processor). I would really appreciate a fix. 

    for idx in range(0, len(Events)):
        inference.append(FindSlices(EyeData[idx], Events[idx], 'Inference'))
        noference.append(FindSlices(EyeData[idx], Events[idx], 'Noference'))
    return inference, noference

#FindSlices returns a dataframe

**Comment by ccordoba12** [comment]
@ryscet, a fix will be released in **2.3.6**, which I plan to release next week.
